// containers/InfoPage/index.js
import React from 'react';
import { Button, CardActions, Card, CardContent, CardMedia, Typography, Divider } from '@material-ui/core';
import {Link, withRouter} from 'react-router-dom';
import {fetchPhonesById} from "actions";
import { connect } from 'react-redux';
const R = require('ramda');

class InfoPage extends React.Component {

    componentWillMount() {
        this.props.fetchPhonesById(this.props.match.params.id)
    }

    render() {
        const phoneById = this.props.storeState.phones[this.props.match.params.id];
        let phone ='';
        if(phoneById) {
            phone = phoneById;
        }
        return(
            <div style={{display: "flex"}}>
                <Card style={{width: "60%", height: "600px", margin: "20px", display: "flex",
                    flexDirection: "row", flexWrap: "wrap"}}>
                    <CardMedia
                        style={{margin: "20px", backgroundSize: "contain", width: "30%"}}
                        image={`${phone.image}`}
                        title={`${phone.name}`}
                    />
                    <div style={{width: "60%", flexDirection: 'column', margin: "20px"}}>
                        <CardContent style={{marginTop: "10%"}}>
                            <Typography>{phone.cpu}</Typography>
                            <Divider/>
                            <Typography>{phone.camera}</Typography>
                            <Divider/>
                            <Typography>{phone.size}</Typography>
                            <Divider/>
                            <Typography>{phone.weight}</Typography>
                            <Divider/>
                            <Typography>{phone.display}</Typography>
                            <Divider/>
                            <Typography>{phone.battery}</Typography>
                            <Divider/>
                            <Typography>{phone.memory}</Typography>
                            <Divider/>
                        </CardContent>
                    </div>
                    <CardContent style={{width: "100%"}}>
                        <Typography style={{margin: "5px 0 5px 0"}}>
                            <Link to={'/'}>
                                <p style={{float: "left", width: "60%"}}>{phone.name}</p>
                                <p style={{float: "left", width: "40%"}}>${phone.price}</p>
                            </Link>
                        </Typography>
                        <Typography>
                            {phone.description}
                        </Typography>
                    </CardContent>
                </Card>
                <Card style={{width: "30%", margin: "20px 10px 0 0", height: "600px"}}>
                    <CardContent>
                        <Typography paragraph={true} variant={'headline'} style={{textAlign: "center"}}>
                            Quick Shop
                        </Typography>
                    </CardContent>
                    <CardActions>
                        <Button
                            fullWidth={true}
                            variant={'contained'}
                            color={'primary'}
                        >
                            Items
                        </Button>
                    </CardActions>
                    <CardContent>
                        <Typography paragraph={true} variant={'display1'}>
                            {phone.name}
                        </Typography>
                        <Typography variant={'headline'}>
                            ${phone.price}
                        </Typography>
                    </CardContent>
                    <CardActions>
                        <Button
                            fullWidth={true}
                            variant={'contained'}
                            color={'primary'}
                        >
                            Add to cart
                        </Button>
                    </CardActions>
                    <CardActions>
                        <Button
                            fullWidth={true}
                            variant={'contained'}
                        >
                            <Link to={"/"}>
                                Back to store
                            </Link>
                        </Button>
                    </CardActions>
                </Card>
            </div>
        );
    }
}

export default withRouter(connect(
    state => ({
        storeState: state
    }),
    dispatch => ({
        fetchPhonesById: (id) => {
            dispatch(fetchPhonesById(id))
        }
    })
)(InfoPage));
